pkgs/applications/misc/snowsql/darwin.nix

```
@@ -0,0 +1,56 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  unzip,
+  cpio,
+  gzip,
+  unar,
+  darwin,
+  zlib,
+  version,
+  meta,
+  pname,
+}: let
+  arch =
+    if stdenv.isAarch64
+    then "arm64"
+    else "x86_64";
+in
+  stdenv.mkDerivation rec {
+    inherit pname version meta;
+    dontStrip = true;
+
+    src = fetchurl {
+      url = "https://sfc-repo.snowflakecomputing.com/snowsql/bootstrap/${lib.versions.majorMinor version}/darwin_${arch}/snowsql-${version}-darwin_${arch}.pkg";
+      sha256 =
+        if arch == "arm64"
+        then "acb2096b87466f0fdbff544d9f64feafc23cbee1bdf963e84b9658511ade9536"
+        else "9727c07fc11b1d8adf4a4eb0b5b996d82cd3d9da191a86f4c7b772726f8e5e92";
```

I think using `stdenvNoCC.isAarch64` is more accurate and future-proof here:
```suggestion
      sha256 =
        if stdenvNoCC.isAarch64
        then "acb2096b87466f0fdbff544d9f64feafc23cbee1bdf963e84b9658511ade9536"
        else "9727c07fc11b1d8adf4a4eb0b5b996d82cd3d9da191a86f4c7b772726f8e5e92";
```